### PR TITLE
PS-123/cloudSQL

### PR DESCRIPTION
**Pull request by danperezsan**
Added CloudSQL composite module for private PostgreSQL resources

### CODE REVIEW

File: gcp/databases/cloudsql/private_postgres/main.tf [hunk 1]
```diff
@@ -0,0 +1,45 @@
+data "google_compute_network" "main" {
+  name = var.vpc_network
+  project = var.project_id
+}
+
+module "private-service-access" {
+  source = "git::https://github.com/terraform-google-modules/terraform-google-sql-db.git//modules/private_service_access?ref=v11.0.0"
```
Comment (AlexBMes) [inline gcp/databases/cloudsql/private_postgres/main.tf:7]: `git::https://` is unneccesary , should be able to do `github.com/terraform-google-modules/terraform-google-sql-db//modules/private_service_access?ref=v11.0.0` for github sources ([docs](https://www.terraform.io/language/modules/sources#github))

File: gcp/databases/cloudsql/private_postgres/versions.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
+terraform {
+  required_version = ">= 0.13"
```
Comment (gbmeuk) [inline gcp/databases/cloudsql/private_postgres/versions.tf:2]: This is old. Has it been ran against the latest version, and are all attributes accounted for?

File: gcp/databases/cloudsql/private_postgres/main.tf [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  project_id  = var.project_id
+  vpc_network = var.vpc_network
+  address = var.private_address
+}
+
+module "cloudsql-postgres" {
+  source = "git::https://github.com/terraform-google-modules/terraform-google-sql-db.git//modules/postgresql?ref=v11.0.0"
```
Comment (AlexBMes) [inline gcp/databases/cloudsql/private_postgres/main.tf:14]: Same as above, could just do `github.com/terraform-google-modules/terraform-google-sql-db//modules/postgresql?ref=v11.0.0`

File: gcp/databases/cloudsql/private_postgres/versions.tf [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.1"
+    }
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 4.4.0, < 5.0"
+    }
+    google-beta = {
```
Comment (gbmeuk) [inline gcp/databases/cloudsql/private_postgres/versions.tf:16]: What needs beta? Add some comment in the code as to why.

File: gcp/databases/cloudsql/private_postgres/main.tf [hunk 5]
```diff
@@ -0,0 +1,45 @@
+data "google_compute_network" "main" {
```
Comment (gbmeuk) [inline gcp/databases/cloudsql/private_postgres/main.tf:1]: Remove HCL1 syntax quotes

File: gcp/databases/cloudsql/private_postgres/versions.tf (hunk 2 above)
Comment (gbmeuk) [reply gcp/databases/cloudsql/private_postgres/versions.tf]: @danperezsan, did this get confirmed?

File: tests/postgres-private-test/main.tf [hunk 6]
```diff
@@ -0,0 +1,38 @@
+module "private-cloudsql-postgres" {
+  source = "../../gcp/databases/cloudsql/private_postgres"
+  project_id  = "cryptotraders-platform-test"
```
Comment (gbmeuk) [inline tests/postgres-private-test/main.tf:3]: Randomise values here.

**Review by gbmeuk — CHANGES_REQUESTED** [review]
Looks fine to me. Just some small points to cover

File: gcp/databases/cloudsql/private_postgres/main.tf (hunk 5 above)
Comment (danperezsan) [reply gcp/databases/cloudsql/private_postgres/main.tf:1]: Removed

File: tests/postgres-private-test/main.tf (hunk 6 above)
Comment (danperezsan) [reply tests/postgres-private-test/main.tf:3]: Done

File: gcp/databases/cloudsql/private_postgres/versions.tf (hunk 4 above)
Comment (danperezsan) [reply gcp/databases/cloudsql/private_postgres/versions.tf]: Not needed

File: gcp/databases/cloudsql/private_postgres/versions.tf (hunk 2 above)
Comment (danperezsan) [reply gcp/databases/cloudsql/private_postgres/versions.tf]: Successfully ran against latest terraform version

**Pull request merged**